plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeMetadata.java

```diff
@@ -465,6 +473,35 @@ public void testGetInputInfoForUnPartitionedTable()
         assertThat(deltaLakeMetadata.getInfo(tableHandle)).isEqualTo(Optional.of(new DeltaLakeInputInfo(false)));
     }
 
+    @Test
+    public void testCreateSchemaTwiceInTheSameSession()
+    {
+        String testSchemaName = "test_schema_" + randomNameSuffix();
+        DeltaLakeMetadata deltaLakeMetadata = deltaLakeMetadataFactory.create(SESSION.getIdentity());
+
+        deltaLakeMetadata.createSchema(SESSION, testSchemaName, Map.of(), TRINO_PRINCIPAL);
+        assertThat(deltaLakeMetadata.schemaExists(SESSION, testSchemaName)).isTrue();
+
+        deltaLakeMetadata.createSchema(SESSION, testSchemaName, Map.of(), TRINO_PRINCIPAL);
+        assertThat(deltaLakeMetadata.schemaExists(SESSION, testSchemaName)).isTrue();
+    }
+
+    @Test
+    public void testCreateSameSchemaFromDifferentSessionsFails()
+    {
+        String testSchemaName = "test_schema_" + randomNameSuffix();
+        DeltaLakeMetadata deltaLakeMetadata = deltaLakeMetadataFactory.create(SESSION.getIdentity());
+
+        deltaLakeMetadata.createSchema(SESSION, testSchemaName, Map.of(), TRINO_PRINCIPAL);
+        assertThat(deltaLakeMetadata.schemaExists(SESSION, testSchemaName)).isTrue();
+
+        ConnectorSession secondSession = TestingConnectorSession.builder().build();
+        assertThatThrownBy(() -> deltaLakeMetadata.createSchema(secondSession, testSchemaName, Map.of(), TRINO_PRINCIPAL))
+                .isInstanceOf(SchemaAlreadyExistsException.class)
+                .hasMessage(format("Schema already exists: '%s'", testSchemaName));
+        assertThat(deltaLakeMetadata.schemaExists(SESSION, testSchemaName)).isTrue();
+    }
```

Comment:
Can we revert this change in this class or add some comments for clarifying the test purpose?